From 7aebd0788a78d388a811ff9ff2b968b31ca0e4fb Mon Sep 17 00:00:00 2001
From: LuYahan <yahan@iscas.ac.cn>
Date: Fri, 16 Jan 2026 11:38:06 +0800
Subject: [PATCH] [riscv] Fix  incorrect code gen

Change-Id: I1f2aa22a5a475b9247e0f26972ebc32d4a42d881
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/7484947
Auto-Submit: Yahan Lu (LuYahan) <yahan@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Yahan Lu (LuYahan) <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#104734}
---
 src/compiler/backend/riscv/instruction-selector-riscv64.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compiler/backend/riscv/instruction-selector-riscv64.cc b/src/compiler/backend/riscv/instruction-selector-riscv64.cc
index 922b2892432..4ba40dff50c 100644
--- a/src/compiler/backend/riscv/instruction-selector-riscv64.cc
+++ b/src/compiler/backend/riscv/instruction-selector-riscv64.cc
@@ -997,7 +997,7 @@ void InstructionSelector::VisitInt32Add(OpIndex node) {
 // Try to match Add(z, a, slli(x, y)) and emit shxadd(z, x, a) for it.
 bool TryEmitShxadd(InstructionSelector* selector, OpIndex add, OpIndex lhs,
                    OpIndex rhs) {
-  if (CpuFeatures::IsSupported(ZBA)) return false;
+  if (!CpuFeatures::IsSupported(ZBA)) return false;
   const Operation& left_op = selector->Get(lhs);
   if ((left_op.Is<Opmask::kWord64ShiftLeft>() &&
        selector->CanCover(add, lhs))) {
-- 
2.52.0

